main/src/mill/main/RunScript.scala

```diff
@@ -319,11 +320,38 @@ object RunScript {
       }
   }
 
+  def evaluateTasks1[T](
+      evaluator: Evaluator,
+      scriptArgs: Seq[String],
+      selectMode: SelectMode
+  ): Either[String, (Seq[PathRef], Either[String, Seq[(Any, Option[(String, ujson.Value)])]])] = {
+    for (targets <- resolveTasks(mill.main.ResolveTasks, evaluator, scriptArgs, selectMode))
+      yield {
+        val (watched, res) = evaluate1(evaluator, Agg.from(targets.distinct))
+
+        val watched2 = for {
+          x <- res.toSeq
+          (Watched(_, extraWatched), _) <- x
+          w <- extraWatched
+        } yield w
+
+        (watched ++ watched2, res)
+      }
+  }
+
   def evaluate(
       evaluator: Evaluator,
       targets: Agg[Task[Any]]
   ): (Seq[PathRef], Either[String, Seq[(Any, Option[ujson.Value])]]) = {
-    val evaluated = evaluator.evaluate(targets)
+    val res = evaluate1(evaluator, targets)
+    (res._1, res._2.map(_.map(p => (p._1, p._2.map(_._2)))))
```

Comment:
You can use pattern matching here:
```suggestion
    val (name, json) = evaluate1(evaluator, targets)
     (name, json.map(_.map(p => (name, json.map(_._2)))))
```

Reply:
The idea is nice, but your snippet don't look correct. The Left of `evaluate1` isn't the name.

Reply:
Yeah, the `suggestion` was just a way to show you what I meant :)